linker/shared/src/main/scala/org/scalajs/linker/backend/BasicLinkerBackend.scala

```diff
@@ -66,6 +66,8 @@ final class BasicLinkerBackend(config: LinkerBackendImpl.Config)
       protected def writeModule(moduleID: ModuleID, jsFileWriter: ByteArrayWriter): Unit = {
         val printedModuleCache = printedModuleSetCache.getModuleCache(moduleID)
 
+        jsFileWriter.sizeHint(sizeHintFor(printedModuleCache.getPreviousFinalJSFileSize()))
```

Comment:
Rather than this heuristic: Have you considered calculating the exact required size for the JS file output?

In incremental runs, only the trees can change and we precompute these as buffers anyways. So you could sum up the exact size (downside is that this will not work for the source maps).

Whether that's faster is probably anyone's guess before we measure it.

Reply:
I considered it but discarded the idea precisely because it wouldn't work for source maps. So we would need two different mechanisms.